android/sandwich/src/main/kotlin/io/qonversion/sandwich/QonversionSandwich.kt

```diff
@@ -365,6 +380,48 @@ class QonversionSandwich(
         return mapOf("success" to success)
     }
 
+    private fun configurePurchaseOptions(
+        offerId: String? = null,
+        applyOffer: Boolean? = null,
+        oldProduct: QProduct? = null,
+        updatePolicyKey: String? = null,
+        contextKeys: List<String>? = null,
+    ): QPurchaseOptions {
+        val builder = QPurchaseOptions.Builder()
+
+        if (oldProduct != null) {
+            builder.setOldProduct(oldProduct)
+        }
+
+        val updatePolicy = updatePolicyKey?.let {
+            try {
+                QPurchaseUpdatePolicy.valueOf(it)
+            } catch (e: IllegalArgumentException) {
+                null
+            }
+        }
+
+        if (updatePolicy != null) {
+            builder.setUpdatePolicy(updatePolicy)
+        }
+
+        if (offerId != null) {
+            builder.setOfferId(offerId)
+        }
+
+        if (applyOffer == false) {
+            builder.removeOffer()
+        }
+
+        if (contextKeys != null) {
+            builder.setContextKeys(contextKeys)
+        }
```

Comment:
I'm voting for more kotlin-way syntax
```suggestion
        oldProduct?.let {
            builder.setOldProduct(it)
        }

        updatePolicyKey?.let {
            try {
                QPurchaseUpdatePolicy.valueOf(it)
            } catch (e: IllegalArgumentException) {
                null
            }
        }?.let {
            builder.setUpdatePolicy(it)
        }

        offerId?.let {
            builder.setOfferId(it)
        }

        if (applyOffer == false) {
            builder.removeOffer()
        }

        contextKeys?.let {
            builder.setContextKeys(it)
        }
```